clipper_admin/clipper_admin/deployers/pytorch.py

```diff
@@ -0,0 +1,191 @@
+        A ``ClipperConnection`` object connected to a running Clipper cluster.
+    name : str
+        The name to be assigned to both the registered application and deployed model.
+    version : str
+        The version to assign this model. Versions must be unique on a per-model
+        basis, but may be re-used across different models.
+    input_type : str
+        The input_type to be associated with the registered app and deployed model.
+        One of "integers", "floats", "doubles", "bytes", or "strings".
+    func : function
+        The prediction function. Any state associated with the function will be
+        captured via closure capture and pickled with Cloudpickle.
+    pytorch_model : pytorch model object
+        The Pytorch model to save.
+    base_image : str, optional
+        The base Docker image to build the new model image from. This
+        image should contain all code necessary to run a Clipper model
+        container RPC client.
+    labels : list(str), optional
+        A list of strings annotating the model. These are ignored by Clipper
+        and used purely for user annotations.
+    registry : str, optional
+        The Docker container registry to push the freshly built model to. Note
+        that if you are running Clipper on Kubernetes, this registry must be accesible
+        to the Kubernetes cluster in order to fetch the container from the registry.
+    num_replicas : int, optional
+        The number of replicas of the model to create. The number of replicas
+        for a model can be changed at any time with
+        :py:meth:`clipper.ClipperConnection.set_num_replicas`.
+    Example
+    -------
+    
+        from clipper_admin import ClipperConnection, DockerContainerManager
+        from clipper_admin.deployers.pytorch import deploy_pytorch_model
+        from torch import nn
+
+        clipper_conn = ClipperConnection(DockerContainerManager())
+
+        # Connect to an already-running Clipper cluster
+        clipper_conn.connect()
+        
+        model = nn.Linear(1,1)
+        
+        #define a shift function to normalize prediction inputs
+        def predict(model, inputs):
+            pred = model(shift(inputs))
+            pred = pred.data.numpy()
+            return [str(x) for x in pred]
+
+        deploy_pytorch_model(
+            clipper_conn,
+            name="example",
+            version = 1,
+            input_type="doubles",
+            func=predict,
+            pytorch_model=model)
+    """
+
+    serialization_dir = save_python_function(name, func)
+
+    # save Torch model
+    torch_weights_save_loc = os.path.join(serialization_dir,
+                                          PYTORCH_WEIGHTS_RELATIVE_PATH)
+
+    torch_model_save_loc = os.path.join(serialization_dir,
+                                        PYTORCH_MODEL_RELATIVE_PATH)
+
+    try:
+        torch.save(pytorch_model.state_dict(), torch_weights_save_loc)
```

Comment:
We now save the model weights and class definition separately. Weights are saved via PyTorch, and the class is pickled with CloudPickle